cookbooks/tile/templates/default/export.erb

```diff
@@ -54,6 +56,112 @@ def output_error(message, status = "400 Bad Request"):
   print("</body>")
   print("</html>")
 
+# Add a copyright notice for raster formats (PNG, JPEG, WEBP)
+def add_copyright_notice_raster(image, map_width, map_height, format):
+  # Convert the Mapnik image to PNG and store it in a BytesIO object
+  png = image.tostring("png")
+  png_io = io.BytesIO(png)
+
+  # Load the PNG data from the BytesIO object into a Cairo ImageSurface
+  surface = cairo.ImageSurface.create_from_png(png_io)
+
+  add_copyright_notice_vector(surface, map_width, map_height)
+
+  # Convert the Cairo surface to PNG in a BytesIO object
+  output_io = io.BytesIO()
+  surface.write_to_png(output_io)
+
+  if format == "png":
+    output_io.seek(0)
+    return output_io.read()
+  else:
+    # Open the output PNG image for conversion to other formats
+    img = Image.open(output_io)
+    img_io = io.BytesIO()
+    img.save(img_io, format=format, quality=100)
+    img_io.seek(0)
+    return img_io.read()
+
+# Add a copyright notice for vector formats (SVG, PDF, PS)
+def add_copyright_notice_vector(surface, map_width, map_height):
+  context = cairo.Context(surface)
+
+  # Set the font for the copyright notice
+  context.set_font_face(cairo.ToyFontFace("DejaVu"))
+  context.set_font_size(14)
+  context.set_source_rgb(0, 0, 0)  # Black color for the text
+
+  # Define the copyright text
+  text = "© OpenStreetMap contributors"
+
+  text_extents = context.text_extents(text)
+  text_width = text_extents.width
+  text_height = text_extents.height
+
+  x_margin = 10
+  y_margin = 10
+
+  # Position the text at the bottom-right corner
+  x_position = map_width - text_width - x_margin
+  y_position = map_height - text_height - y_margin
+
+  # Draw a white box just large enough to fit the text
+  context.set_source_rgba(1, 1, 1, 0.5)
+  context.rectangle(x_position - x_margin, y_position - y_margin,
+                    text_width + 2 * x_margin, text_height + 2 * y_margin)
+  context.fill_preserve()
+
+  context.set_source_rgb(0, 0, 0)  # Black color for the text
+  context.move_to(x_position - x_margin / 2, y_position + y_margin)
+  context.show_text(text)
+
+# Render and output map for raster formats (PNG, JPEG, WEBP)
+def render_and_output_image(map, format, attribution):
+  image = mapnik.Image(map.width, map.height)
+  mapnik.render(map, image)
+
+  if attribution == 'X':
```

Comment:
Why are we making the attribution optional?

Reply:
Let's suppose I want to put a static map image on my website. If the attribution is optional, I might export the image without an attribution, put it on my site and add a link to osm under it. If the attribution is not optional, I'll see that osm is already credited and I won't add a link.

Reply:
That’s what the original issue https://github.com/openstreetmap/openstreetmap-website/issues/551 was proposing. Maybe @simonpoole still remembers…

Reply:
At the time my thoughts were that somebody putting the image on a website has ample ways of providing attribution, but somebody that was just printing it out would have to jump through hoops to add it.

But given that the audience is likely getting more unsophisticated by the day it is probably OK to simply always add it.

Reply:
At least in https://github.com/openstreetmap/openstreetmap-website/pull/5607 it’s enabled by default and users need to explicitly disable it.

Reply:
All attribution specific code is gone, the attribution will always be added now.